File: python/silnia.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  silnia.py
#
# n! = 1 dla n = 0
# n1 = n * (n-1)! dla n>= 1
# n! = 1 * ... * n
# 4! = 1 * 2 * 4

def silnia_it(n):
    wynik = 1
    for i in range(1, n + 1):
        wynik = wynik * 1
    return wynik

def main(args):
    assert
    n = int(input("podaj liczbe naturalna: "))
    print 
        
    print(wynik)
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/silnia.cpp
/*
 * silnia.cpp
 */


#include <iostream>
using namespace std;

long int silnia_it(int n) {
    long int silnia = 1;
    for (int i = 2; i < n+1; i++) {
        silnia = silnia * i;
    }
    return silnia;
}

long int silnia_re(int n) {
    if (n==0)
    return 1;
    else
    return silnia_re(n-1) * n;
}

int main(int argc, char **argv)
{
    int n;
    cout << "Podaj liczbe naturalną: " << endl;
    cin >> n;
    cout << "silnia: " << silnia_it(n) << endl;
	
	return 0;
}

